### PR TITLE
feat:  Add support for GET published entries endpoint [SPA-817]

### PR DESCRIPTION
**Pull request by Inbal-Gordon**
##  Description

The purpose of this PR is to bring back the visibility of all published versions of the entries for CMA by adding support for GET published entries endpoint.

## Motivation and Context

A user asked this on [github](https://github.com/contentful/contentful-management.js/issues/1419#issue-1286007785) as once this functionality was available 

## Checklist (check all before merging)

- [x] Both unit and integration tests are passing
- [x] There are no breaking changes
- [x] Changes are reflected in the documentation (will be on a separate PR)

When adding a new method:

- [x] The new method is exported through the default and plain CMA client
- [x] All new public types are exported from `./lib/export-types.ts`
- [x] Added a unit test for the new method
- [x] Added an integration test for the new method
- [x] The new method is added to the documentation

### CODE REVIEW

**Review by Spring3 — CHANGES_REQUESTED** [review]
Good start. I pointed to some of the mistakes made which caused the type errors

Also please update the name of the PR to better reflect what is being done. `chore: try to fix types` is a good name for a commit, but the PR does more than fixing types.

Same for the type of the branch. `feat` would be better because we are adding a new feature here. `chore` doesn't work becuase when you merge it, it will not bump the version of the package

File: lib/plain/plain-client.ts [hunk 1]
```diff
@@ -160,6 +160,7 @@ export const createPlainClient = (
       delete: wrap(wrapParams, 'Task', 'delete'),
     },
     entry: {
+      getPublishedEntries: wrap(wrapParams, 'Entry', 'getPublishedEntries'),
```
Comment (Spring3) [inline lib/plain/plain-client.ts:163]: same here. Please rename it back to `getPublished`

File: lib/create-environment-api.ts [hunk 2]
```diff
@@ -719,6 +719,39 @@ export default function createEnvironmentApi(makeRequest: MakeRequest) {
...
+     * .then((environment) => environment.getPublishedEntries{'content_type': 'foo'})) // you can add more queries as 'key': 'value'
+     * .then((response) => console.log(response.items))
+     * .catch(console.error)
+     * ```
+     */
+    getPublishedEntries(query: QueryOptions = {}) {
+      const raw = this.toPlainObject() as EnvironmentProps
+      return makeRequest({
+        entityType: 'Entry',
+        action: 'getMany',
```
Comment (Spring3) [inline lib/create-environment-api.ts:746]: Here it is calling the function called `getMany` while we need to call the function called `getPublished` which you have added before

File: lib/adapters/REST/endpoints/entry.ts [hunk 3]
```diff
@@ -30,6 +30,22 @@ export const get: RestEndpoint<'Entry', 'get'> = <T extends KeyValueMap = KeyVal
   )
 }
 
+export const getPublishedEntries:RestEndpoint<'Entry', 'getPublishedEntries'> = <T extends KeyValueMap = KeyValueMap>(
```
Comment (Spring3) [inline lib/adapters/REST/endpoints/entry.ts:33]: here we can keep the old name - `getPublished` because users will call it as `entry.getPublished` and it will already be clear that they are requesting entries

with the current name they would call it as `entry.getPublishedEntries` so the word `entry` appears twice and it is unnecessary

**Review by Spring3 — DISMISSED** [review]

File: lib/common-types.ts [hunk 4]
```diff
@@ -599,6 +600,7 @@ export interface Adapter {
  */
 export type MRActions = {
   Http: {
+    queryPublished: { params: { url: string; config?: AxiosRequestConfig }; return: any }
```
[the remark this answers is not in the document]
Comment (Spring3) [reply lib/common-types.ts]: [Here I updated the pointer to the correct section](https://github.com/contentful/contentful-management.js/blob/2264d1e39658442c6d1ea405f090d3f343514764/lib/common-types.ts#L946)

**Comment by Spring3** [comment]
Also, don't forget to add the type for your new function [here](https://github.com/contentful/contentful-management.js/blob/2264d1e39658442c6d1ea405f090d3f343514764/lib/common-types.ts#L372) for `Entry`

File: test/helpers.js [hunk 5]
```diff
@@ -73,8 +73,12 @@ export async function getTestUser() {
   return organization.getUser(userId)
 }
 
-export async function getDefaultSpace() {
+export async function getDefaultSpace(plainClient = false) {
   const { spaceId } = TestDefaults
+
+  if (plainClient) {
```
Comment (SofiaMargariti) [inline test/helpers.js:79]: Why was this change needed here?

**Comment by SofiaMargariti** [comment]
I've noticed package-lock.json has a lot of changes in this PR. Was that on purpose? How come this PR resulted in so many changes if we didn't update any dependencies?

**Comment by Inbal-Gordon** [comment]
@SofiaMargariti the reason for this is because I was mistakenly asked by Dan to remove my package-json file and use yarn however the original project on github uses package-json and npm not yarn

File: test/helpers.js (hunk 5 above)
Comment (luizfonseca) [reply test/helpers.js:79]: to create a different client base on the default space, though @Inbal-Gordon we can probably improve it:

- We can undo the change on this `helpers.js`
- In `entry-integration` `line 413` we can use ` plainClient = initPlainClient({ spaceId: TestDefaults.spaceId })` instead. Make sure the imports are working.

File: test/integration/entry-integration.js [hunk 6]
```diff
@@ -395,4 +401,20 @@ describe('Entry Api', () => {
...
     })
   })
+
+  describe('read plainClientApi', () => {
+    /**
+     * @type {import('../../lib/contentful-management').PlainClientAPI}
+     */
+    let plainClient
+    before(async () => {
+      plainClient = await getDefaultSpace(true)
```
Comment (luizfonseca) [inline test/integration/entry-integration.js:411]:
```suggestion
       plainClient = initPlainClient({ spaceId: TestDefaults.spaceId })
```

so we don't need that change in `helpers.js`

**Comment by luizfonseca** [comment]
@Inbal-Gordon  Code looks good, just some minor comments. Let us know once they are addressed.

**Comment by SofiaMargariti** [comment]
> @SofiaMargariti the reason for this is because I was mistakenly asked by Dan to remove my package-json file and use yarn however the original project on github uses package-json and npm not yarn

I see, it looks like it's due to using a newer version of npm, but it looks like it is backwards compatible so it should be ok 👍

**Comment by Inbal-Gordon** [comment]
@SofiaMargariti @luizfonseca I have updated the test and removed the change we did in the helpers.js file

**Review by luizfonseca — APPROVED** [review]

File: lib/adapters/REST/endpoints/entry.ts [hunk 7]
```diff
@@ -30,6 +30,24 @@ export const get: RestEndpoint<'Entry', 'get'> = <T extends KeyValueMap = KeyVal
...
+  T extends KeyValueMap = KeyValueMap
+>(
+  http: AxiosInstance,
+  params: GetSpaceEnvironmentParams & QueryParams,
+  rawData?: unknown,
+  headers?: AxiosRequestHeaders
+) => {
+  return raw.get<CollectionProp<EntryProps<T>>>(
+    http,
+    `/spaces/${params.spaceId}/environments/${params.environmentId}/public/entries`,
```
Comment (Inbal-Gordon) [inline lib/adapters/REST/endpoints/entry.ts:43]: @luizfonseca Is the path with `/public` correct here?
